tensorflow_datasets/core/custom_dataset/translate_dataset.py

```diff
@@ -0,0 +1,135 @@
+"""Custom Translate Datasets Template"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+from typing import Dict, List, Tuple, Set
+
+import tensorflow.compat.v2 as tf
+from tensorflow_datasets.core import dataset_builder
+from tensorflow_datasets.core import dataset_info
+from tensorflow_datasets.core import features as features_lib
+from tensorflow_datasets.core import splits as split_lib
+from tensorflow_datasets.core.utils import version
+
+class TranslateDataset(dataset_builder.DatasetBuilder):
+  """Generic image classification dataset created from manual directory.
+
+  The data directory should have the following structure:
+
+  ```
+  path/to/manual_dir/<dataset_name>/
+    split_name/  # Ex: 'train'
+      lang1.txt
+      lang2.txt
+    split_name/  # Ex: 'test'
+      lang1.txt
+      lang2.txt
+      ...
+  ```
+
+  To use it:
+
+  ```
+  builder = tfds.TranslateDataset(root_dir='path/to/manual_dir/')
+  print(builder.info)  # Splits, num examples,... are automatically calculated
+  ds = builder.as_dataset(split='train', shuffle_files=True)
+  ```
+
+  """
+
+  VERSION = version.Version('1.0.0')
+
+  def __init__(self, root_dir: str):
+    root_dir = os.path.expanduser(root_dir)
+    self._languages_examples_dict, languages = _get_language_examples(root_dir)
+    self._languages = sorted(list(languages))
```

Comment:
You only need the languages in `__init__`. No need to load all files (e.g. loading the train set if only the eval set is used).

You could only extract dict `split -> filepaths` and load the required files in `_as_datasets`

Reply:
We will have to read all the file content to update the split info, right?

Reply:
I see. I guess it's fine to load the full data then.